lib/internal/readline/promises.js

```diff
@@ -20,13 +21,18 @@ const {
 } = CSI;
 
 class Readline {
+  #autoCommit = false;
   #stream;
   #todo = [];
 
-  constructor(stream) {
+  constructor(stream, options = ObjectCreate(null)) {
     if (!isWritable(stream))
       throw new ERR_INVALID_ARG_TYPE('stream', 'Writable', stream);
     this.#stream = stream;
+    if (options.autoCommit != null) {
```

Comment:
```suggestion
  constructor(stream, options = undefined) {
    if (!isWritable(stream))
      throw new ERR_INVALID_ARG_TYPE('stream', 'Writable', stream);
    this.#stream = stream;
    if (options?.autoCommit != null) {
```
that way you don't need to make a throwaway object, and you also won't throw if someone passes `null` here